src/client/src/routes/StyleguideRoute/styled/mapMarginPaddingProps.ts

```diff
@@ -0,0 +1,95 @@
+import _ from 'lodash'
+import { css } from 'rt-theme'
+
+import { mapProps } from '../tools'
+
+// export type PropValue = number | string | boolean
+export type PropValue = any
+
+export interface MarginProps {
+  m?: PropValue
+  mx?: PropValue | 'auto'
+  my?: PropValue
+  mt?: PropValue
+  mr?: PropValue | 'auto'
+  mb?: PropValue
+  ml?: PropValue | 'auto'
+}
+
+export interface PaddingProps {
+  p?: PropValue
+  px?: PropValue
+  py?: PropValue
+  pt?: PropValue
+  pr?: PropValue
+  pb?: PropValue
+  pl?: PropValue
+}
+
+export interface MarginPaddingProps extends MarginProps, PaddingProps {}
+
+export const { marginPaddingProps, mapMarginPaddingProps } = (() => {
```

Comment:
Some comments might help the user understand what you're trying to accomplish.

Reply:
I will add a comment at the top to this documentation — in general anything that gets closer to `styled-system` for this kind of primitive is ideal.

https://jxnblk.com/styled-system/getting-started#margin--padding